source/common/runtime/runtime_features.cc

```diff
@@ -110,7 +110,8 @@ FALSE_RUNTIME_GUARD(envoy_reloadable_features_runtime_initialized);
 FALSE_RUNTIME_GUARD(envoy_reloadable_features_always_use_v6);
 // TODO(pradeepcrao) reset this to true after 2 releases (1.27)
 FALSE_RUNTIME_GUARD(envoy_reloadable_features_enable_include_histograms);
-
+// TODO(krajshiva) reset this to true after 2 releases (1.28)
+FALSE_RUNTIME_GUARD(envoy_reloadable_features_enable_sanitization_during_sink);
```

Comment:
A possible more general solution (I would iterate in a doc) is to make this runtime flag be a list of string-replacements to make in all stat names. It could default to the current ones in the sanitizer function which (as you've observed) may not really be related to statsd despite the comment and original intent of that function.

Reply:
Then you could make the statsd-specific sanitizer really just implement whatever statsd requires and nothing else.

Reply:
I think its fine for sanitize_all_stats_for_statsd (default: true) to reflect current problematic behavior (hijack of sanitization by statsd format). Will make the change.